cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -157,19 +168,50 @@ protected EventHandler createEventHandler(CConfiguration cConf) {
     return new AbortOnTimeoutEventHandler(cConf.getLong(Constants.CFG_TWILL_NO_CONTAINER_TIMEOUT, Long.MAX_VALUE));
   }
 
+  // returns the path of the file, if its a local URI. If it's an HDFS URI, copy it to local file system first.
```

Comment:
can't use that, because it doesn't allow make the new file to have restricted permissions.